Instagram/View Controllers/ComposeViewController.m

```diff
@@ -12,35 +12,63 @@ @interface ComposeViewController () <UIImagePickerControllerDelegate, UINavigati
 
 @property (weak, nonatomic) IBOutlet UIImageView *imageView;
 @property (weak, nonatomic) IBOutlet UITextField *captionField;
+@property (weak, nonatomic) IBOutlet UIProgressView *progressView;
+@property (weak, nonatomic) IBOutlet UILabel *progressLabel;
 
 @end
 
 @implementation ComposeViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    [self hideProgress];
 }
 
 - (void)clear {
     UIImage *placeholderImage = [UIImage imageNamed:@"image_placeholder"];
     [self.imageView setImage:placeholderImage];
     [self.captionField setText:@""];
+    [self hideProgress];
     [self dismissViewControllerAnimated:YES completion:nil];
 }
 
 - (void)imagePickerController:(UIImagePickerController *)picker
 didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
     
     UIImage *editedImage = info[UIImagePickerControllerEditedImage];
-    [self.imageView setImage:editedImage];
+    UIImage *resizedImage = [self compress:editedImage];
+    [self.imageView setImage:resizedImage];
     [self dismissViewControllerAnimated:YES completion:nil];
 }
 
+- (UIImage *)compress:(UIImage *)image {
+    NSInteger maxBytes = 10000000; // 10 MB
+    CGFloat compressionQuality = 1.0;
```

Comment:
Protip: If you put `f` at the end, the literal will be a float literal and won't need to be cast from a double. I think even a lot of professional iOS engineers don't necessarily know/do this and it's not clear whether it makes any real difference in performance, but it's something that I've gotten used to doing.

i.e. `1.0f` or `1.f`